File: 백준/Silver/1874. 스택 수열/스택 수열.java
import java.io.*;
import java.util.*;
public class Main {
    public static void main(String[] args) throws IOException{
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();
        Stack<Integer> st = new Stack<>();

        int n = Integer.parseInt(br.readLine());
        int i = 1;
        boolean flag = true;

        for(int s = 0; s < n; s++){
            int num = Integer.parseInt(br.readLine());
            if(i<=num){
                while(i<=num){
                    st.push(i++);
                    sb.append("+\n");
                }
                st.pop();
                sb.append("-\n");
            }else{
                if(st.pop()>num){
                    System.out.println("NO");
                    flag = false;
                    break;
                }
                sb.append("-\n");
            }
        }
        if (flag) System.out.println(sb);
    }
}

File: 백준/Silver/1874. 스택 수열/스택 수열.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.lang.StringBuilder
import java.util.*

fun main(){
    val br = BufferedReader(InputStreamReader(System.`in`))
//    val bw = BufferedWriter(OutputStreamWriter(System.out))
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    var i = 1
    var flag = true
    val st = Stack<Int>()

    for (s in 0..<n){
        val num = br.readLine().toInt()
        if (i <= num){
            while (i <= num){
                st.push(i)
                sb.append("+\n")
                i++
            }
            st.pop()
            sb.append("-\n")
        }else {
            val op = st.pop()
            if(op>num){
                print("NO")
                flag =false
                break
            }else{
                sb.append("-\n")
            }
        }
    }
    if (flag){
//        bw.flush()
//        bw.close()
        println(sb)
    }
}